CMakeLists.txt

```diff
@@ -61,6 +61,7 @@ option(BUILD_PYTHON_BINDINGS "Build Python bindings with pybind11." OFF)
 option(BUILD_TESTING_PYTHON "Build Python tests only." OFF)
 
 if (BUILD_PYTHON_BINDINGS)
+  set(PYBIND11_FINDPYTHON ON)
```

Comment:
According to [the doc](https://pybind11.readthedocs.io/en/stable/cmake/index.html#modes), this is a CMake 3.12 (3.15 recommended) feature. `manif` support goes back to Ubuntu 18.04 which [ships with CMake 3.10.2](https://packages.ubuntu.com/search?keywords=cmake).

Reply:
Right, I missed this. I guess in this case I'm going to revert to the classic mode. The only problem is that I need to get the location of python interpreter. I'll figure out what is the best way with CMake 3.12.

An alternative I used in other projects is vendoring a modern FindPython files in the project, so that the behavior is consistent among all OSs and CMake versions. If you prefer proceeding this way, I can update this PR accordingly.

Reply:
Btw, I just noticed that in Ubuntu 16.04, python2 is found. Not sure if this is the intended behavior:

https://github.com/diegoferigo/manif/runs/2839281906?check_suite_focus=true#step:5:85

Reply:
> An alternative I used in other projects is vendoring a modern FindPython files in the project, so that the behavior is consistent among all OSs and CMake versions. If you prefer proceeding this way, I can update this PR accordingly.

I'd actually prefer to vendor as little as possible since that means more things to maintain. Especially true with things I'm not familiar with.

Reply:
> > An alternative I used in other projects is vendoring a modern FindPython files in the project, so that the behavior is consistent among all OSs and CMake versions. If you prefer proceeding this way, I can update this PR accordingly.
> 
> I'd actually prefer to vendor as little as possible since that means more things to maintain. Especially true with things I'm not familiar with.

I understand it, no problem. Right now, in order to debug the failing test, I vendored the resources in dff4930. This removes from the picture possible problems of CMake detection. It seems that the error in CI occurrs only with Python 3.6, if I update the pybind11 job to 3.8 it succeeds. Not yet sure why, maybe a linked library is missing.

Anyway, the trade-off between vendoring or using plain CMake, is that old CMake version have either poor or no Python support at all. CMake 3.10, for instance, only has [`FindPythonInterp`](https://cmake.org/cmake/help/v3.10/module/FindPythonInterp.html) that in my experience can create many mismatch in the found Python version. Though, if you prefer dealing with possible issues rather than maintaining the vendored files, I will update the logic to support plain CMake 3.10.

Reply:
> Though, if you prefer dealing with possible issues rather than maintaining the vendored files

Well when you put it that way I'm not sure I really have a choice :sweat_smile:. 
Again, my concern is that I'm not familiar with all that and I care about being able to maintain the feature without too much pain.

Reply:
In the past I had to deal with a lot of related issues and I'm biased about it :) The vendored files come directly from the [CMake tree](https://github.com/Kitware/CMake/tree/v3.20.4/Modules), and unless major changes (which I doubt will happen) these vendored resources can stay as they are for years. I'd not be very concerned about their maintenance.

I managed to solve the problems in CI by passing `Python3_EXECUTABLE` as was done before. Even though in other projects I didn't have the need to use this hint, in this project for some reason it's the only way to have a proper detection from `setup.py` for all the supported platforms (ubuntu and macos) and all the python versions (3.6 - 3.9).